src/number_theory/cyclotomic/basic.lean

```diff
@@ -95,6 +95,37 @@ lemma iff_singleton : is_cyclotomic_extension {n} A B ↔
  (∀ x, x ∈ adjoin A { b : B | b ^ (n : ℕ) = 1 }) :=
 by simp [is_cyclotomic_extension_iff]
 
+/-- Given `(f : B ≃ₐ[A] C)`, if `is_cyclotomic_extension S A B` then
+`is_cyclotomic_extension S A C`. -/
+@[protected] lemma equiv {C : Type*} [comm_ring C] [algebra A C] [h : is_cyclotomic_extension S A B]
```

Comment:
Can't you reuse `trans` below? Then you would only need to show that an equiv is an `is_cyclotomic_extension {1}`, I guess.

Reply:
That's a good idea. We need that `is_cyclotomic_extension (S ∪ {1}) _ _ ↔ is_cyclotomic_extension S _ _`, but I can even prove something more general (replacing `1` by a divisors of every element in `S`). I will do it in the next days.

Reply:
It's clashing with `module.finite`. There is something strange, if I don't open `module` then it's clashing with `is_cyclotomic_extension.finite` even if I've protected it. (I just pushed this version, it seems better anyway)